.github/workflows/docker-build-push.yml

```diff
@@ -21,20 +31,63 @@ jobs:
             type=ref,event=tag
 
       - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v1.7.0
+        uses: docker/setup-buildx-action@v2.0.0
 
       - name: Login to DockerHub
         if: github.event_name != 'pull_request'
-        uses: docker/login-action@v1.14.1
+        uses: docker/login-action@v2.0.0
         with:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_TOKEN }}
 
       - name: Build image and push to Docker Hub
-        uses: docker/build-push-action@v2.10.0
+        uses: docker/build-push-action@v3.0.0
         with:
           context: .
           file: docker/Dockerfile
           push: ${{ github.event_name != 'pull_request' }}
           tags: ${{ steps.docker_metadata.outputs.tags }}
           labels: ${{ steps.docker_metadata.outputs.labels }}
+
+  deploy_to_gce:
+    name: Deploy the image to a GCE VM
+    runs-on: ubuntu-latest
+    needs: push_to_registry
+
+    # Add "id-token" with the intended permissions.
+    permissions:
+      contents: "read"
+      id-token: "write"
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      # Authentication via credentials json
+      - id: "auth"
+        uses: "google-github-actions/auth@v0"
+        with:
+          credentials_json: "${{ secrets.GCE_SA_KEY }}"
+
+      # Setup gcloud CLI
+      - name: Set up Cloud SDK
+        uses: google-github-actions/setup-gcloud@v0
+
+      # Make sure the VM is up
+      - name: Start the deploy-pudl-vm
+        run: gcloud compute instances start "$GCE_INSTANCE" --zone="$GCE_INSTANCE_ZONE"
+
+      # Deploy PUDL image to GCE
+      - name: Deploy
+        run: |-
+          gcloud compute instances update-container "$GCE_INSTANCE" \
+            --zone "$GCE_INSTANCE_ZONE" \
+            --container-image "docker.io/${{needs.push_to_registry.outputs.docker_tags}}" \
+            --container-command "conda" \
+            --container-arg="run" \
+            --container-arg="--no-capture-output" \
+            --container-arg="-p" \
+            --container-arg="/home/catalyst/env" \
+            --container-arg="bash" \
+            --container-arg="./docker/pudl_etl.sh" \
+            --container-env-file="./docker/.env" \
```

Comment:
How does the instance get shut down when all this has completed?

Reply:
I need to add some code to the dockerfile that calls the google API to shut the instance down. I haven't included it yet because I like sshing into the VM to review the VM and container logs. I can add the code and comment it out for now.

Reply:
Ah okay so it's not shutting itself down yet.  Yeah adding it in and commenting it out just as a placeholder / reminder would be good.  Is that like a "post command" clean up section?

Reply:
Ok. I was going to add it to `pudl_etl.sh`

Reply:
Have you come across any best practices on how this is supposed to work? Putting a suicide command inside the script seems a little weird.